src/core/window/clipboard.js

```diff
@@ -44,13 +51,60 @@ export function copyTextToClipboard(win, text) {
 
   removeElement(textarea);
 
-  return copySuccessful;
+  if (copySuccessful) {
+    successCallback();
+  } else {
+    failCallback();
+  }
 }
 
 /**
- * @param {Document} doc
- * @return {boolean}
+ * @param {Document} doc Document context
+ * @return {boolean} Returns true if copying support is available with deprecated API
  */
-export function isCopyingToClipboardSupported(doc) {
+export function isDeprecatedCopyingToClipboardSupported(doc) {
   return doc.queryCommandSupported('copy');
 }
+
+/**
+ * Copies provided text to clipboard using 'navigator.clipboard' API.
+ * It will try to use deprecated API if 'navigator.clipboard' is not supported.
+ * @param {Window} win Window context
+ * @param {string} text Text to copy
+ * @param {*} successCallback Executes when copying is successful
+ * @param {*} failCallback Executes when copying is failed
+ */
+export function copyTextToClipboard(win, text, successCallback, failCallback) {
+  /** Check which method is supported for the browser */
+  if (!!win.navigator?.clipboard) {
+    /*
+      Try copying with `navigator.clipboard` method as a fallback support.
+    */
+    win.navigator.clipboard.writeText(text).then(
+      function () {
+        /* Clipboard successfully set using 'navigator.clipboard' */
+        successCallback();
+      },
+      function () {
+        /* Clipboard write failed using 'navigator.clipboard' */
+        failCallback();
+      }
```

Comment:
Can we just pass the callsbacks directly to `then`?
```suggestion
      successCallback(),
      failCallback()
```

Reply:
no parens if passed directly
```suggestion
        successCallback, failCallback
```

Reply:
Thank you for highlighting. I have corrected these calls to callback.